src/docfx/build/Build.cs

```diff
@@ -11,41 +11,53 @@ namespace Microsoft.Docs.Build
 {
     internal static class Build
     {
-        public static async Task Run(string docsetPath, CommandLineOptions options, ErrorLog errorLog)
+        public static async Task Run(string docsetPath, CommandLineOptions options)
         {
-            // load and trace entry repository
-            var repositoryProvider = new RepositoryProvider(docsetPath, options);
-            var repository = repositoryProvider.GetRepository(FileOrigin.Default);
-            Telemetry.SetRepository(repository?.Remote, repository?.Branch);
-            var locale = LocalizationUtility.GetLocale(repository, options);
+            List<Error> errors;
+            Config config = null;
 
-            using (var restoreGitMap = RestoreGitMap.Create(docsetPath, locale))
+            using (var errorLog = new ErrorLog("build", docsetPath, options.Output, () => config, options.Legacy))
             {
-                // load configuration from current docset and fallback docset
-                var input = new Input(docsetPath, repositoryProvider);
-                var configLoader = new ConfigLoader(docsetPath, input, repositoryProvider);
-
-                repositoryProvider.ConfigRestoreMap(restoreGitMap);
-                var (errors, config) = configLoader.Load(options, extend: true);
-
-                errorLog.Configure(config);
-
-                // just return if config loading has errors
-                if (errorLog.Write(errors))
-                    return;
+                try
+                {
+                    // load and trace entry repository
+                    var repositoryProvider = new RepositoryProvider(docsetPath, options);
+                    var repository = repositoryProvider.GetRepository(FileOrigin.Default);
+                    Telemetry.SetRepository(repository?.Remote, repository?.Branch);
+                    var locale = LocalizationUtility.GetLocale(repository, options);
 
-                // get docsets(build docset, fallback docset and dependency docsets)
-                repositoryProvider.Config(config);
-                var (docset, fallbackDocset) = GetDocsetWithFallback(docsetPath, locale, config, repositoryProvider, restoreGitMap);
-                if (!string.Equals(docset.DocsetPath, PathUtility.NormalizeFolder(docsetPath), PathUtility.PathComparison))
+                    using (var restoreGitMap = RestoreGitMap.Create(docsetPath, locale))
+                    {
+                        // load configuration from current docset and fallback docset
+                        var input = new Input(docsetPath, repositoryProvider);
+                        var configLoader = new ConfigLoader(docsetPath, input, repositoryProvider);
+
+                        repositoryProvider.ConfigRestoreMap(restoreGitMap);
+                        (errors, config) = configLoader.Load(options, extend: true);
```

Comment:
yes, and there are 2 cases:

1. command line args contains `--output`, we'll create an `.errors.log` file under `--output` path.
2. there is no `--output`, errors will be shown in console only since we don't know where to write `.errors.log` file.